tpot/metrics.py

```diff
@@ -45,10 +45,12 @@ def balanced_accuracy(y_true, y_pred):
     all_classes = list(set(np.append(y_true, y_pred)))
     all_class_accuracies = []
     for this_class in all_classes:
-        this_class_sensitivity = \
-            float(sum((y_pred == this_class) & (y_true == this_class))) /\
-            float(sum((y_true == this_class)))
-
+        if float(sum((y_true == this_class))) == 0:
+            this_class_sensitivity = 0
+        else:
+            this_class_sensitivity = \
+                float(sum((y_pred == this_class) & (y_true == this_class))) /\
+                float(sum((y_true == this_class)))
         this_class_specificity = \
             float(sum((y_pred != this_class) & (y_true != this_class))) /\
             float(sum((y_true != this_class)))
```

Comment:
You are correct. This would be another scenario that could cause an division by zero error where the true label are all identical and this identical label is among predicted labels. For example:
```
y_true_test = np.array([1, 1, 1, 1])
y_pred_test = np.array([1, 2, 3, 4])
```
The updated test case is here [test updated.zip](https://github.com/rhiever/tpot/files/1072241/test.updated.zip) and all tests passed. I'll update the branch.

> 
> Result on [0, 1, 2, 3] and [0, 1, 2, 3]:
> Original:
> 1.0
> Fixed:
> 1.0
> Result on [0, 1, 2, 3] and [1, 1, 2, 3]:
> Original:
> 0.833333333333
> Fixed:
> 0.833333333333
> Result on [0, 1, 2, 3] and [1, 2, 3, 4]:
> Original:
> Division by zero in this_class_sensitivity
> Fixed:
> 0.375
> Result on [0, 1, 2, 3] and [1, 2, 3, 4]:
> Original:
> Division by zero in this_class_specificity
> Fixed:
> 0.3125
> 
> ----------------------------------------------------------------------
> Ran 97 tests in 35.367s
> 
> OK